<div align="center">

**LAPORAN PRAKTIKUM**

**ALGORITMA & PEMROGRAMAN**

**"OPERATOR PADA PYTHON"**

**Disusun Oleh :**

**Raldo Restanggi Alfarizy**

**2611533018**

**Dosen Pengampu:**

**Dr. Wahyudi, S.T., M.T.**

**Asisten Praktikum:**

**Sherly Sukmadira**

<br>

<img src="images/logo_unand.png" width="200">

<br>

**FAKULTAS TEKNOLOGI INFORMASI**

**DEPARTEMEN INFORMATIKA**

**UNIVERSITAS ANDALAS**

**2026**

</div>

## KATA PENGANTAR

Puji syukur penulis panjatkan atas kehadirat Tuhan Yang Maha Esa yang telah memberikan rahmat dan berkat-Nya, sehingga penulis dapat menyelesaikan laporan praktikum Operator pada Python yang telah dilaksanakan sebagai pertemuan ke-3 untuk mata kuliah Praktikum Algoritma dan Pemrograman. Praktikum ini berfokus pada pembahasan jenis-jenis operator dalam bahasa Python, yaitu operator aritmatika, perbandingan, logika, penugasan, keanggotaan, identitas, serta bitwise, beserta penerapannya dalam sebuah studi kasus terpadu.

Melalui praktikum ini, hasil yang diharapkan adalah penulis dapat memahami dan mengimplementasikan secara langsung konsep operator Python pada sebuah program nyata, yaitu sistem simulasi transaksi dan validasi akses toko.

Penulis menyadari bahwa laporan praktikum ini masih memiliki banyak kekurangan. Oleh karena itu, kritik dan saran yang membangun sangat diharapkan demi perbaikan dan pengembangan di masa mendatang.

Padang, 17 September 2026

---
# BAB I
# PENDAHULUAN

## 1.1 Latar Belakang
Operator merupakan simbol khusus yang digunakan untuk melakukan operasi terhadap satu atau lebih nilai (operand) di dalam sebuah program. Hampir seluruh proses pengolahan data pada program, mulai dari perhitungan sederhana hingga pengambilan keputusan yang kompleks, dibangun dari kombinasi operator. Tanpa pemahaman yang baik mengenai operator, seorang pemrogram akan kesulitan menerjemahkan logika permasalahan ke dalam bentuk kode.

Python menyediakan beragam jenis operator, mulai dari operator aritmatika untuk perhitungan matematis, operator perbandingan dan logika untuk pengambilan keputusan, operator penugasan untuk memperbarui nilai variabel, operator keanggotaan dan identitas untuk pemeriksaan data, hingga operator bitwise yang bekerja langsung pada level bit. Penguasaan seluruh jenis operator tersebut menjadi dasar penting sebelum mempelajari struktur kontrol, perulangan, dan struktur data yang lebih lanjut.

Pada praktikum pekan ini, konsep operator diterapkan pada studi kasus sistem simulasi transaksi dan validasi akses toko. Studi kasus ini dipilih karena mampu menggabungkan seluruh jenis operator dalam satu alur program yang utuh: perhitungan total pembayaran menggunakan operator aritmatika, pemeriksaan syarat diskon menggunakan operator perbandingan dan logika, pemeriksaan kode promo menggunakan operator keanggotaan, serta penentuan hak akses pelanggan menggunakan operator bitwise.

## 1.2 Tujuan
1. Memahami jenis-jenis operator yang tersedia pada bahasa pemrograman Python beserta fungsinya masing-masing.
2. Mampu menerapkan operator aritmatika, perbandingan, logika, penugasan, keanggotaan, identitas, dan bitwise ke dalam sebuah program.
3. Mampu membedakan penggunaan operator `is` dengan operator `==` pada Python.
4. Mampu merancang satu program terpadu yang mengintegrasikan seluruh jenis operator untuk menyelesaikan sebuah studi kasus.

## 1.3 Manfaat
1. Penulis dapat menerjemahkan logika sebuah permasalahan nyata ke dalam bentuk ekspresi operator pada Python.
2. Penulis memahami representasi kondisi dalam bentuk bit sehingga dapat digunakan untuk pengelolaan hak akses secara efisien.
3. Penulis terbiasa menyusun program yang menerima masukan dari pengguna dan menampilkan keluaran yang terstruktur.

---
# BAB II
# PEMBAHASAN

## 2.1 Dasar Teori

Operator adalah simbol yang memberi tahu interpreter untuk melakukan operasi tertentu terhadap operand. Pada Python, operator dikelompokkan menjadi beberapa jenis sebagai berikut.

1. **Operator Aritmatika**, digunakan untuk melakukan operasi matematika. Terdiri atas penjumlahan (`+`), pengurangan (`-`), perkalian (`*`), pembagian (`/`), pembagian bulat (`//`), sisa bagi atau modulus (`%`), dan pangkat (`**`). Perlu diperhatikan bahwa operator `/` selalu menghasilkan bilangan bertipe float, sedangkan `//` membulatkan hasil ke bawah menjadi bilangan bulat.

2. **Operator Perbandingan**, digunakan untuk membandingkan dua buah nilai dan selalu menghasilkan nilai boolean `True` atau `False`. Terdiri atas `==`, `!=`, `>`, `<`, `>=`, dan `<=`.

3. **Operator Logika**, digunakan untuk menggabungkan dua atau lebih kondisi boolean. Terdiri atas `and` (bernilai benar jika seluruh kondisi benar), `or` (bernilai benar jika salah satu kondisi benar), dan `not` (membalik nilai kebenaran suatu kondisi).

4. **Operator Penugasan**, digunakan untuk memberikan nilai kepada variabel. Selain operator dasar `=`, terdapat *augmented assignment* seperti `+=`, `-=`, `*=`, `/=`, `//=`, `%=`, dan `**=` yang menggabungkan operasi aritmatika dengan penugasan sehingga penulisan kode menjadi lebih ringkas.

5. **Operator Keanggotaan (Membership)**, yaitu `in` dan `not in`, digunakan untuk memeriksa keberadaan suatu nilai di dalam objek sekuens seperti list, tuple, atau string.

6. **Operator Identitas (Identity)**, yaitu `is` dan `is not`, digunakan untuk memeriksa apakah dua variabel merujuk pada objek yang sama di dalam memori. Operator ini berbeda dengan `==` yang membandingkan nilai atau isi objek.

7. **Operator Bitwise**, digunakan untuk melakukan operasi pada level bit dari bilangan bulat. Terdiri atas AND (`&`), OR (`|`), XOR (`^`), NOT (`~`), geser kiri (`<<`), dan geser kanan (`>>`).

Selain itu, pada praktikum ini juga digunakan fungsi `input()` untuk menerima masukan dari pengguna. Nilai yang dikembalikan `input()` selalu bertipe string, sehingga perlu dikonversi menggunakan `int()` apabila akan digunakan dalam operasi aritmatika.

## 2.2 Kode Program (Latihan Praktikum)

Bagian ini mendokumentasikan seluruh latihan praktikum pekan 3. Setiap sel kode dijalankan langsung pada notebook ini sehingga kode beserta outputnya tersimpan menjadi satu kesatuan dan tidak memerlukan tangkapan layar terpisah. Karena program menggunakan fungsi `input()`, saat sel dijalankan ulang pengguna perlu mengetikkan kembali masukan pada kotak input yang muncul.

### 2.2.1 Program "aritmatika_2611533018.py"

In [2]:
# Buat file dengan nama aritmatika_2611533018.py
# Buat program untuk operator aritmatika dalam python
# Nama variabel ditambah 4 digit nim terakhir contoh: angka_3018
# Program ini menggunakan fungsi input ()
# Nilai yang dimasukkan akan dikonversi menjadi tipe data integer

angka1_3018 = int(input("Masukkan angka-1: "))
angka2_3018 = int(input("Masukkan angka-2: "))

# Penjumlahan
hasil_penjumlahan_3018 = angka1_3018 + angka2_3018
print("\nOperator Penjumlahan")
print("Hasil =", hasil_penjumlahan_3018)

# Pengurangan
hasil_pengurangan_3018 = angka1_3018 - angka2_3018
print("\nOperator Pengurangan")
print("Hasil =", hasil_pengurangan_3018)

# Perkalian
hasil_perkalian_3018 = angka1_3018 * angka2_3018
print("\nOperator Perkalian")
print("Hasil =", hasil_perkalian_3018)

# Pembagian, pembagian bulat, dan sisa bagi
if angka2_3018 != 0:
    hasil_pembagian_3018 = angka1_3018 / angka2_3018
    print("\nOperator Pembagian")
    print("Hasil =", hasil_pembagian_3018)

    hasil_pembagian_bulat_3018 = angka1_3018 // angka2_3018
    print("\nOperator Pembagian bulat")
    print("Hasil =", hasil_pembagian_bulat_3018)

    hasil_sisa_bagi_3018 = angka1_3018 % angka2_3018
    print("\nOperator sisa bagi")
    print("Hasil =", hasil_sisa_bagi_3018)
else:
    print("Angka kedua tidak boleh bernilai nol")

# Pangkat
hasil_pangkat_3018 = angka1_3018 ** angka2_3018
print("\nOperator Pangkat")
print("Hasil =", hasil_pangkat_3018)



Operator Penjumlahan
Hasil = 19

Operator Pengurangan
Hasil = 11

Operator Perkalian
Hasil = 60

Operator Pembagian
Hasil = 3.75

Operator Pembagian bulat
Hasil = 3

Operator sisa bagi
Hasil = 3

Operator Pangkat
Hasil = 50625


**Penjelasan :**
1. Program meminta dua buah bilangan bulat dari pengguna menggunakan fungsi `input()` yang hasilnya dikonversi dengan `int()` agar dapat dihitung.
2. Kedua bilangan tersebut dioperasikan dengan tujuh operator aritmatika, yaitu `+`, `-`, `*`, `/`, `//`, `%`, dan `**`. Setiap hasil disimpan pada variabel `hasil_3018` lalu ditimpa oleh operasi berikutnya.
3. Setiap hasil dicetak bersama ekspresi perhitungannya. Pada contoh eksekusi terlihat `15 / 4` menghasilkan `3.75` bertipe float, sedangkan `15 // 4` menghasilkan `3` karena dibulatkan ke bawah, dan `15 % 4` menghasilkan sisa `3`.

### 2.2.2 Program "perbandingan_2611533018.py"

In [4]:
# Buat file dengan nama perbandingan_2611533018.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka_3018
# Program ini menggunakan fungsi input ()
# Nilai yang dimasukkan akan dikonversi menjadi tipe data integer
# Program operator perbandingan dalam Python

angka1_3018 = int(input("Masukkan angka-1: "))
angka2_3018 = int(input("Masukkan angka-2: "))

# Lebih besar dari
hasil_lebihbesar_3018 = angka1_3018 > angka2_3018
print("\nOperator lebih besar dari")
print("angka1 > angka2 =", hasil_lebihbesar_3018)

# Lebih kecil dari
hasil_lebihkecil_3018 = angka1_3018 < angka2_3018
print("\nOperator lebih kecil dari")
print("angka1 < angka2 =", hasil_lebihkecil_3018)

# Lebih besar atau sama dengan
hasil_lebihbesarsama_3018 = angka1_3018 >= angka2_3018
print("\nOperator lebih besar dari atau sama dengan")
print("angka1 >= angka2 =", hasil_lebihbesarsama_3018)

# Lebih kecil atau sama dengan
hasil_lebihkecilsama_3018 = angka1_3018 <= angka2_3018
print("\nOperator lebih kecil dari atau sama dengan")
print("angka1 <= angka2 =", hasil_lebihkecilsama_3018)

# Sama dengan
hasil_samadengan_3018 = angka1_3018 == angka2_3018
print("\nOperator sama dengan")
print("angka1 == angka2 =", hasil_samadengan_3018)

# Tidak sama dengan
hasil_tidaksama_3018 = angka1_3018 != angka2_3018
print("\nOperator tidak sama dengan")
print("angka1 != angka2 =", hasil_tidaksama_3018)

# Tambahan : perbandingan berantai dalam python
hasil_perbandingan_berantai_angka1_3018 = 0 < angka1_3018 < 100
print("\nPerbandingan berantai")
print("0 < angka1 < 100 =", hasil_perbandingan_berantai_angka1_3018)

hasil_perbandingan_berantai_angka2_3018 = 0 < angka2_3018 < 100
print("0 < angka2 < 100 =", hasil_perbandingan_berantai_angka2_3018)


Operator lebih besar dari
angka1 > angka2 = True

Operator lebih kecil dari
angka1 < angka2 = False

Operator lebih besar dari atau sama dengan
angka1 >= angka2 = True

Operator lebih kecil dari atau sama dengan
angka1 <= angka2 = False

Operator sama dengan
angka1 == angka2 = False

Operator tidak sama dengan
angka1 != angka2 = True

Perbandingan berantai
0 < angka1 < 100 = True
0 < angka2 < 100 = True


**Penjelasan :**
1. Program menerima dua buah bilangan sebagai masukan pengguna.
2. Kedua bilangan dibandingkan menggunakan enam operator perbandingan, yaitu `==`, `!=`, `>`, `<`, `>=`, dan `<=`.
3. Seluruh hasil perbandingan bertipe boolean. Pada contoh eksekusi dengan angka `12` dan `7`, operator `>` dan `>=` menghasilkan `True`, sedangkan `<`, `<=`, dan `==` menghasilkan `False`.

### 2.2.3 Program "logika_2611533018.py"

In [5]:
# Buat file dengan nama logika_2611533018.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka_3018
# Program ini menggunakan fungsi input ()
# Program operator logika dalam Python

# Masukkan nilai boolean
# Input tidak peka terhadap huruf besar atau kecil
a1_3018 = input("Input nilai boolean-1 (true/false): ").strip().lower() == "true"
a2_3018 = input("Input nilai boolean-2 (true/false): ").strip().lower() == "true"

print("\nA1 =", a1_3018)
print("A2 =", a2_3018)

# Konjungsi: bernilai true jika keduanya true
hasil_konjungsi_3018 = a1_3018 and a2_3018
print("\nKonjungsi (AND)")
print("A1 and A2 = ", hasil_konjungsi_3018)

# Disjungsi: bernilai true jika salah satunya true
hasil_disjungsi_3018 = a1_3018 or a2_3018
print("\nDisjungsi (OR)")
print("A1 or A2 = ", hasil_disjungsi_3018)

# Negasi A1: membalik nilai A1
hasil_negasi_A1_3018 = not a1_3018
print("\nNegasi A1 (NOT)")
print("not A1 = ", hasil_negasi_A1_3018)

# Negasi A2: membalik nilai A2
hasil_negasi_A2_3018 = not a2_3018
print("\nNegasi A2 (NOT)")
print("not A2 = ", hasil_negasi_A2_3018)

# XOR: bernilai true jika kedua nilai berbeda
hasil_xor_3018 = a1_3018 != a2_3018
print("\nDisjungsi Eksklusif (XOR)")
print("A1 XOR A2 = ", hasil_xor_3018)



A1 = True
A2 = False

Konjungsi (AND)
A1 and A2 =  False

Disjungsi (OR)
A1 or A2 =  True

Negasi A1 (NOT)
not A1 =  False

Negasi A2 (NOT)
not A2 =  True

Disjungsi Eksklusif (XOR)
A1 XOR A2 =  True


**Penjelasan :**
1. Program meminta dua kondisi boolean melalui operator perbandingan, yaitu `true` dan `false`.
2. Kedua kondisi tersebut digabungkan menggunakan operator `and` dan `or`, serta dibalik menggunakan operator `not`.
3. Pada contoh eksekusi dengan nilai `true` dan `false`, operator `and` menghasilkan `False` karena kedua nilai input tidak sama, sedangkan `or` menghasilkan `True` karena kedua nilai input berbeda. Hal ini memperlihatkan dengan jelas perbedaan perilaku kedua operator tersebut.

### 2.2.4 Program "assignment_2611533018.py"

In [6]:
# Buat file dengan nama assignment_2611533018.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka_3018
# Program ini menggunakan fungsi input ()
# Nilai yang dimasukkan akan dikonversi menjadi tipe data integer
# Program operator assignment dalam Python

angka1_3018 = int(input("Masukkan angka-1: "))
angka2_3018 = int(input("Masukkan angka-2: "))

print("\nNilai awal angka1 =", angka1_3018)
print("\nNilai awal angka2 =", angka2_3018)

# Assignment biasa
hasil_biasa_3018 = angka1_3018
print("\nHasil assignment biasa (=)")
print("Hasil =", hasil_biasa_3018)

# Assignment penambahan
hasil_penambahan_3018 = angka1_3018
hasil_penambahan_3018 += angka2_3018
print("\nHasil assignment penambahan (+=)")
print("Hasil =", hasil_penambahan_3018)

# Assignment pengurangan
hasil_pengurangan_3018 = angka1_3018
hasil_pengurangan_3018 -= angka2_3018
print("\nHasil assignment pengurangan (-=)")
print("Hasil =", hasil_pengurangan_3018)

# Assignment perkalian
hasil_perkalian_3018 = angka1_3018
hasil_perkalian_3018 *= angka2_3018
print("\nHasil assignment perkalian (*=)")
print("Hasil =", hasil_perkalian_3018)

# Assignment pembagian, bilangan bulat, dan sisa bagi
if angka2_3018 != 0:
    hasil_pembagian_3018 = angka1_3018
    hasil_pembagian_3018 /= angka2_3018
    print("\nHasil assignment pembagian (/=)")
    print("Hasil =", hasil_pembagian_3018)

    hasil_pembagian_bulat_3018 = angka1_3018
    hasil_pembagian_bulat_3018 //= angka2_3018
    print("\nHasil assignment pembagian bulat (//=)")
    print("Hasil =", hasil_pembagian_bulat_3018)

    hasil_sisa_bagi_3018 = angka1_3018
    hasil_sisa_bagi_3018 %= angka2_3018
    print("\nHasil assignment sisa bagi (%=)")
    print("Hasil =", hasil_sisa_bagi_3018)
else:
    print("\nPembagian tidak dapat dilakukan")
    print("Angka kedua tidak boleh bernilai 0.")

# Operator tambahan: assignment pemangkatan
hasil_pangkat_3018 = angka1_3018
hasil_pangkat_3018 **= angka2_3018
print("\nHasil assignment pemangkatan(**=)")
print("Hasil =", hasil_pangkat_3018)


Nilai awal angka1 = 30

Nilai awal angka2 = 25

Hasil assignment biasa (=)
Hasil = 30

Hasil assignment penambahan (+=)
Hasil = 55

Hasil assignment pengurangan (-=)
Hasil = 5

Hasil assignment perkalian (*=)
Hasil = 750

Hasil assignment pembagian (/=)
Hasil = 1.2

Hasil assignment pembagian bulat (//=)
Hasil = 1

Hasil assignment sisa bagi (%=)
Hasil = 5

Hasil assignment pemangkatan(**=)
Hasil = 8472886094430000000000000000000000000


**Penjelasan :**
1. Program memberikan nilai awal pada variabel `angka1_3018` dan `angka2_3018` menggunakan operator penugasan dasar `=`.
2. Nilai variabel tersebut diperbarui secara berurutan menggunakan *augmented assignment* `+=`, `-=`, `*=`, `/=`, `//=`, dan `%=`. Setiap operasi bekerja terhadap nilai terakhir variabel, bukan terhadap nilai awal.
3. Sebelum operasi `**=`, nilai variabel dikembalikan ke nilai awal agar hasil pemangkatan mudah diperiksa. Perlu diperhatikan bahwa setelah operator `/=` digunakan, tipe data variabel berubah menjadi float meskipun nilainya bulat.

### 2.2.5 Program "lainnya_2611533018.py" (Operator Keanggotaan dan Identitas)

In [7]:
# Buat file dengan nama lainnya_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menggunakan fungsi input()
# Program operator keanggotaan dan  identitas
 
print("========================================")
print("1. OPERATOR KEANGGOTAAN")
print("========================================")
 
# Input beberapa data yang dipisahkan dengan koma
input_data_3018 = input("Masukkan beberapa angka, pisahkan dengan koma: ")
 
# Mengubah input menjadi list integer
data_3018 = [int(angka.strip()) for angka in input_data_3018.split(",")]
 
nilai_dicari_3018 = int(input("Masukkan angka yang ingin dicari: "))
 
# Operator in
hasil_3018 = nilai_dicari_3018 in data_3018
print("\nOperator keanggotaan IN")
print(nilai_dicari_3018, "in", data_3018, "=", hasil_3018)
 
# Operator not in
hasil_3018 = nilai_dicari_3018 not in data_3018
print("\nOperator keanggotaan NOT IN")
print(nilai_dicari_3018, "not in", data_3018, "=", hasil_3018)

1. OPERATOR KEANGGOTAAN

Operator keanggotaan IN
23 in [2, 23, 43, 23, 54, 23, 13, 54, 32, 1] = True

Operator keanggotaan NOT IN
23 not in [2, 23, 43, 23, 54, 23, 13, 54, 32, 1] = False


**Penjelasan :**
1. Program meminta pengguna memasukkan beberapa angka yang dipisahkan dengan tanda koma, lalu string masukan dipecah menggunakan `split(",")`. Setiap potongan dibersihkan dengan `strip()` dan dikonversi menjadi integer melalui *list comprehension*, sehingga terbentuk sebuah list bilangan bulat.
2. Bagian pertama memeriksa keberadaan sebuah angka di dalam list menggunakan operator `in` dan `not in`. Hasil pemeriksaan berupa nilai boolean yang saling berlawanan.
3. Pada contoh eksekusi dengan dicari nya bilangan `23` didalam `data_3018`, operator `in` menghasilkan `true` karena bilangan `23` ada pada himpunan di `data_3018`, sedangkan `not in` menghasilkan `false`. Hal ini memperlihatkan dengan jelas perbedaan hasil dari kedua operator tersebut.

### 2.2.6 Program "bitwise_2611533018.py"

In [8]:
# Buat file dengan nama bitwise_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menggunakan fungsi input()
 
print("\n========================================")
print("3. OPERATOR BITWISE")
print("========================================")
 
angka1_3018 = int(input("Masukkan angka bitwise-1: "))
angka2_3018 = int(input("Masukkan angka bitwise-2: "))
 
print("\nAngka dalam bentuk desimal dan biner")
print("angka1 =", angka1_3018, "| biner =", bin(angka1_3018))
print("angka2 =", angka2_3018, "| biner =", bin(angka2_3018))
 
# Bitwise AND
hasil_3018 = angka1_3018 & angka2_3018
print("\nBitwise AND (&)")
print(angka1_3018, "&", angka2_3018, "=", hasil_3018)
print("Biner hasil =", bin(hasil_3018))
print("Biner hasil (8 bit) =", format(hasil_3018, "08b"))
 
# Bitwise OR
hasil_3018 = angka1_3018 | angka2_3018
print("\nBitwise OR (|)")
print(angka1_3018, "|", angka2_3018, "=", hasil_3018)
print("Biner hasil =", bin(hasil_3018))
print("Biner hasil (8 bit) =", format(hasil_3018, "08b"))
 
# Bitwise XOR
hasil_3018 = angka1_3018 ^ angka2_3018
print("\nBitwise XOR (^)")
print(angka1_3018, "^", angka2_3018, "=", hasil_3018)
print("Biner hasil =", bin(hasil_3018))
print("Biner hasil (8 bit) =", format(hasil_3018, "08b"))
 
# Bitwise NOT
hasil_3018 = ~angka1_3018
print("\nBitwise NOT (~)")
print("~", angka1_3018, "=", hasil_3018)
print("Biner hasil =", bin(hasil_3018))
print("Biner hasil (8 bit) =", format(hasil_3018, "08b"))
 
# Bitwise geser kiri
jumlah_geser_3018 = int(input("\nMasukkan jumlah pergeseran bit: "))
 
hasil_3018 = angka1_3018 << jumlah_geser_3018
print("\nBitwise geser kiri (<<)")
print(angka1_3018, "<<", jumlah_geser_3018, "=", hasil_3018)
print("Biner hasil =", bin(hasil_3018))
print("Biner hasil (8 bit) =", format(hasil_3018, "08b"))
 
# Bitwise geser kanan
hasil_3018 = angka1_3018 >> jumlah_geser_3018
print("\nBitwise geser kanan (>>)")
print(angka1_3018, ">>", jumlah_geser_3018, "=", hasil_3018)
print("Biner hasil =", bin(hasil_3018))
print("Biner hasil (8 bit) =", format(hasil_3018, "08b"))


3. OPERATOR BITWISE

Angka dalam bentuk desimal dan biner
angka1 = 12 | biner = 0b1100
angka2 = 10 | biner = 0b1010

Bitwise AND (&)
12 & 10 = 8
Biner hasil = 0b1000
Biner hasil (8 bit) = 00001000

Bitwise OR (|)
12 | 10 = 14
Biner hasil = 0b1110
Biner hasil (8 bit) = 00001110

Bitwise XOR (^)
12 ^ 10 = 6
Biner hasil = 0b110
Biner hasil (8 bit) = 00000110

Bitwise NOT (~)
~ 12 = -13
Biner hasil = -0b1101
Biner hasil (8 bit) = -0001101

Bitwise geser kiri (<<)
12 << 3 = 96
Biner hasil = 0b1100000
Biner hasil (8 bit) = 01100000

Bitwise geser kanan (>>)
12 >> 3 = 1
Biner hasil = 0b1
Biner hasil (8 bit) = 00000001


**Penjelasan :**
1. Program meminta dua bilangan bulat, lalu menampilkan bentuk desimal dan binernya menggunakan fungsi `bin()`.
2. Kedua bilangan dioperasikan dengan operator bitwise AND (`&`), OR (`|`), XOR (`^`), dan NOT (`~`). Setiap hasil ditampilkan dalam bentuk desimal, biner penuh, serta biner 8 bit melalui `format(hasil, "08b")`. Pada contoh eksekusi dengan angka `12` (1100) dan `10` (1010), operasi AND menghasilkan `8` (1000) karena hanya bit yang sama-sama bernilai 1 yang dipertahankan.
3. Bagian terakhir meminta jumlah pergeseran bit, kemudian melakukan geser kiri (`<<`) dan geser kanan (`>>`). Geser kiri sebanyak n bit setara dengan mengalikan bilangan dengan 2 pangkat n, sedangkan geser kanan setara dengan membaginya. Perlu diperhatikan bahwa hasil operator NOT bernilai negatif, sehingga `format(hasil, "08b")` menampilkan tanda minus di depan bentuk binernya.

---
## 2.3 Tugas Praktikum

**Soal :**

Buatlah sebuah program kasir sederhana bernama `tugas3_2611533018.py` yang menghitung total transaksi pelanggan sekaligus memeriksa status pelanggan dan hak aksesnya terhadap promo tertentu. Program menerima data nama pelanggan, status pelanggan, total belanja, jumlah barang, dan kode promo, kemudian menggunakan seluruh jenis operator Python (aritmatika, perbandingan, logika, penugasan, keanggotaan, identitas, dan bitwise) untuk menentukan total pembayaran, status diskon, kelayakan promo, serta hak akses pelanggan dalam satu alur program yang utuh.

Ketentuan penting:
- Setiap variabel diakhiri 4 digit terakhir NIM, yaitu `_3018`.
- Kondisi pelanggan direpresentasikan dalam bit: `0001` member, `0010` belanja minimal Rp200.000, `0100` jumlah barang minimal 3, dan `1000` kode promo tersedia.
- Program wajib menggunakan masukan dari pengguna dan dapat dijalankan tanpa error.

**Kode Jawaban :**

In [9]:
# ==============================================================
# Nama    : Raldo Restanggi Alfarizy
# NIM     : 2611533018
# Kelas   : Praktikum Algoritma dan Pemrograman - Pekan 3
# Topik   : Operator Python
# Studi Kasus : Sistem Simulasi Transaksi dan Validasi Akses Toko
# Catatan : Seluruh nama variabel diakhiri 4 digit terakhir NIM (3018)
# ==============================================================

print("=== SISTEM TRANSAKSI TOKO ===")

# --------------------------------------------------------------
# 1. INPUT DATA PELANGGAN DAN TRANSAKSI
# --------------------------------------------------------------
nama_3018 = input("\nMasukkan Nama Pelanggan : ")
status_3018 = input("Masukkan Status Pelanggan (member/nonmember) : ").strip().lower()
total_belanja_3018 = int(input("Masukkan Total Belanja : "))
jumlah_barang_3018 = int(input("Masukkan Jumlah Barang : "))
kode_promo_3018 = input("Masukkan Kode Promo : ").strip().upper()

# --------------------------------------------------------------
# 2. KONSTANTA / DATA ACUAN TOKO
# --------------------------------------------------------------
batas_belanja_3018 = 200000          # syarat minimum belanja
minimal_barang_3018 = 3              # syarat minimum jumlah barang
persen_diskon_3018 = 10              # besar diskon member (dalam persen)
daftar_promo_3018 = ["HEMAT10", "HEMAT20", "GRATISONGKIR"]

# --------------------------------------------------------------
# 3. OPERATOR PERBANDINGAN  ( >= , == )
#    Menghasilkan nilai bertipe boolean (True / False)
# --------------------------------------------------------------
syarat_belanja_3018 = total_belanja_3018 >= batas_belanja_3018   # operator >=
syarat_barang_3018 = jumlah_barang_3018 >= minimal_barang_3018   # operator >=
is_member_3018 = status_3018 == "member"                         # operator ==

# --------------------------------------------------------------
# 4. OPERATOR KEANGGOTAAN  ( in , not in )
#    Memeriksa apakah kode promo terdapat pada daftar promo toko
# --------------------------------------------------------------
promo_tersedia_3018 = kode_promo_3018 in daftar_promo_3018           # operator in
promo_tidak_ada_3018 = kode_promo_3018 not in daftar_promo_3018      # operator not in

# --------------------------------------------------------------
# 5. OPERATOR LOGIKA  ( and , or , not )
# --------------------------------------------------------------
dapat_diskon_3018 = is_member_3018 and syarat_belanja_3018        # operator and
dapat_promo_3018 = promo_tersedia_3018 and syarat_barang_3018     # operator and
pelanggan_prioritas_3018 = dapat_diskon_3018 or dapat_promo_3018  # operator or
bukan_member_3018 = not is_member_3018                            # operator not

# --------------------------------------------------------------
# 6. OPERATOR ARITMATIKA  ( * , // , - , % )
# --------------------------------------------------------------
if dapat_diskon_3018:
    diskon_3018 = total_belanja_3018 * persen_diskon_3018 // 100   # operator * dan //
else:
    diskon_3018 = 0

# --------------------------------------------------------------
# 7. OPERATOR PENUGASAN  ( = , -= , += , *= )
# --------------------------------------------------------------
total_bayar_3018 = total_belanja_3018      # operator penugasan dasar ( = )
total_bayar_3018 -= diskon_3018            # augmented assignment ( -= )

poin_3018 = 0
poin_3018 += jumlah_barang_3018 * 10       # augmented assignment ( += )
if is_member_3018:
    poin_3018 *= 2                         # augmented assignment ( *= )

# Perhitungan lanjutan (aritmatika): rata-rata dan sisa pembagian
if jumlah_barang_3018 > 0:
    rata_rata_3018 = total_bayar_3018 // jumlah_barang_3018   # pembagian bulat ( // )
    sisa_bagi_3018 = total_bayar_3018 % jumlah_barang_3018    # sisa bagi ( % )
else:
    rata_rata_3018 = 0
    sisa_bagi_3018 = 0

# --------------------------------------------------------------
# 8. OPERATOR IDENTITAS  ( is , is not )
#    salinan_promo_3018  -> menunjuk objek list yang SAMA
#    duplikat_promo_3018 -> objek BARU dengan isi yang sama
# --------------------------------------------------------------
salinan_promo_3018 = daftar_promo_3018        # referensi ke objek yang sama
duplikat_promo_3018 = daftar_promo_3018[:]    # objek baru hasil penyalinan

identitas_sama_3018 = salinan_promo_3018 is daftar_promo_3018        # operator is
identitas_beda_3018 = duplikat_promo_3018 is not daftar_promo_3018   # operator is not
nilai_sama_3018 = duplikat_promo_3018 == daftar_promo_3018           # operator ==

# --------------------------------------------------------------
# 9. OPERATOR BITWISE  ( | , & , ^ , << )
#    Setiap kondisi pelanggan diwakili satu bit:
#    0001 = member | 0010 = belanja >= 200000
#    0100 = barang >= 3 | 1000 = kode promo tersedia
# --------------------------------------------------------------
bit_member_3018 = 0b0001
bit_belanja_3018 = 0b0010
bit_barang_3018 = 0b0100
bit_promo_3018 = 0b1000

kode_status_3018 = 0
if is_member_3018:
    kode_status_3018 |= bit_member_3018      # operator OR ( | )
if syarat_belanja_3018:
    kode_status_3018 |= bit_belanja_3018     # operator OR ( | )
if syarat_barang_3018:
    kode_status_3018 |= bit_barang_3018      # operator OR ( | )
if promo_tersedia_3018:
    kode_status_3018 |= bit_promo_3018       # operator OR ( | )

cek_member_3018 = kode_status_3018 & bit_member_3018     # operator AND ( & )
cek_promo_3018 = kode_status_3018 & bit_promo_3018       # operator AND ( & )

kode_referensi_3018 = 0b1011                             # kode transaksi pembanding
beda_status_3018 = kode_status_3018 ^ kode_referensi_3018  # operator XOR ( ^ )

geser_kiri_3018 = kode_status_3018 << 1                  # operator geser kiri ( << )

# Hak akses pelanggan ditentukan dari pemeriksaan bit
mask_ongkir_3018 = bit_barang_3018 | bit_promo_3018      # 1100
member_access_3018 = cek_member_3018 == bit_member_3018
promo_access_3018 = cek_promo_3018 == bit_promo_3018
free_shipping_3018 = (kode_status_3018 & mask_ongkir_3018) == mask_ongkir_3018

# ==============================================================
# 10. TAMPILAN HASIL PROGRAM
# ==============================================================
print()
print("=== DATA PELANGGAN ===")
print("Nama Pelanggan       :", nama_3018)
print("Status Pelanggan     :", status_3018)
print("Total Belanja        : Rp" + str(total_belanja_3018))
print("Jumlah Barang        :", jumlah_barang_3018)
print("Kode Promo           :", kode_promo_3018)

print()
print("=== HASIL PERHITUNGAN ===")
print("Besar Diskon         : Rp" + str(diskon_3018))
print("Total Pembayaran     : Rp" + str(total_bayar_3018))
print("Rata-rata Harga      : Rp" + str(rata_rata_3018))
print("Sisa Pembagian (%)   :", sisa_bagi_3018)
print("Poin Pelanggan       :", poin_3018)

print()
print("=== HASIL VALIDASI ===")
print("Belanja >= Rp" + str(batas_belanja_3018) + "   :", syarat_belanja_3018)
print("Jumlah Barang >= " + str(minimal_barang_3018) + "     :", syarat_barang_3018)
print("Status Member          :", is_member_3018)
print("Kode Promo Tersedia    :", promo_tersedia_3018)
print("Mendapatkan Diskon     :", dapat_diskon_3018)
print("Mendapatkan Promo      :", dapat_promo_3018)
print("Pelanggan Prioritas    :", pelanggan_prioritas_3018)

print()
print("=== HAK AKSES PELANGGAN ===")
print("Kode Hak Akses (biner) :", format(kode_status_3018, "04b"))
print("Kode Hak Akses (des)   :", kode_status_3018)
print("Member Access          :", member_access_3018)
print("Promo Access           :", promo_access_3018)
print("Free Shipping Access   :", free_shipping_3018)

print()
print("=== HASIL OPERATOR ===")

print()
print("-- Operator Aritmatika --")
print("Diskon  :", total_belanja_3018, "*", persen_diskon_3018, "// 100 =", diskon_3018)
print("Bayar   :", total_belanja_3018, "-", diskon_3018, "=", total_bayar_3018)
print("Rata2   :", total_bayar_3018, "//", jumlah_barang_3018, "=", rata_rata_3018)
print("Sisa    :", total_bayar_3018, "%", jumlah_barang_3018, "=", sisa_bagi_3018)

print()
print("-- Operator Perbandingan --")
print(total_belanja_3018, ">=", batas_belanja_3018, "=", syarat_belanja_3018)
print(jumlah_barang_3018, ">=", minimal_barang_3018, "=", syarat_barang_3018)
print("'" + status_3018 + "'", "==", "'member'", "=", is_member_3018)

print()
print("-- Operator Logika --")
print("member and belanja cukup (and) =", dapat_diskon_3018)
print("dapat diskon or dapat promo (or) =", pelanggan_prioritas_3018)
print("not member (not) =", bukan_member_3018)

print()
print("-- Operator Penugasan --")
print("total_bayar -= diskon  -> Rp" + str(total_bayar_3018))
print("poin += jumlah*10      -> ", jumlah_barang_3018 * 10)
print("poin *= 2 (jika member)-> ", poin_3018)

print()
print("-- Operator Keanggotaan --")
print("Daftar promo toko :", daftar_promo_3018)
print("'" + kode_promo_3018 + "' in daftar_promo =", promo_tersedia_3018)
print("'" + kode_promo_3018 + "' not in daftar_promo =", promo_tidak_ada_3018)

print()
print("-- Operator Identitas --")
print("salinan is daftar_promo      =", identitas_sama_3018)
print("duplikat is not daftar_promo =", identitas_beda_3018)
print("duplikat == daftar_promo     =", nilai_sama_3018)
print("Kesimpulan: 'is' membandingkan identitas objek, '==' membandingkan nilai/isi objek.")

print()
print("-- Operator Bitwise --")
print("Kode Status Transaksi")
print("Kode Biner   :", format(kode_status_3018, "04b"))
print("Kode Desimal :", kode_status_3018)

print()
print("Cek Member")
print(format(kode_status_3018, "04b"), "&", format(bit_member_3018, "04b"))
print("Hasil Biner   :", format(cek_member_3018, "04b"))
print("Hasil Desimal :", cek_member_3018)

print()
print("Cek Promo")
print(format(kode_status_3018, "04b"), "&", format(bit_promo_3018, "04b"))
print("Hasil Biner   :", format(cek_promo_3018, "04b"))
print("Hasil Desimal :", cek_promo_3018)

print()
print("Perbandingan Status")
print("Kode Transaksi :", format(kode_status_3018, "04b"))
print("Kode Referensi :", format(kode_referensi_3018, "04b"))
print(format(kode_status_3018, "04b"), "^", format(kode_referensi_3018, "04b"))
print("Hasil Biner   :", format(beda_status_3018, "04b"))
print("Hasil Desimal :", beda_status_3018)

print()
print("Shift")
print(format(kode_status_3018, "04b"), "<< 1")
print("Hasil Biner   :", format(geser_kiri_3018, "05b"))
print("Hasil Desimal :", geser_kiri_3018)

print()
print("=== SELESAI ===")


=== SISTEM TRANSAKSI TOKO ===

=== DATA PELANGGAN ===
Nama Pelanggan       : Raldo Restanggi Alfarizy
Status Pelanggan     : member
Total Belanja        : Rp250000
Jumlah Barang        : 4
Kode Promo           : HEMAT10

=== HASIL PERHITUNGAN ===
Besar Diskon         : Rp25000
Total Pembayaran     : Rp225000
Rata-rata Harga      : Rp56250
Sisa Pembagian (%)   : 0
Poin Pelanggan       : 80

=== HASIL VALIDASI ===
Belanja >= Rp200000   : True
Jumlah Barang >= 3     : True
Status Member          : True
Kode Promo Tersedia    : True
Mendapatkan Diskon     : True
Mendapatkan Promo      : True
Pelanggan Prioritas    : True

=== HAK AKSES PELANGGAN ===
Kode Hak Akses (biner) : 1111
Kode Hak Akses (des)   : 15
Member Access          : True
Promo Access           : True
Free Shipping Access   : True

=== HASIL OPERATOR ===

-- Operator Aritmatika --
Diskon  : 250000 * 10 // 100 = 25000
Bayar   : 250000 - 25000 = 225000
Rata2   : 225000 // 4 = 56250
Sisa    : 225000 % 4 = 0

-- Operator Perbandi

**Penjelasan Alur Program :**

Output di atas merupakan hasil eksekusi program dengan masukan nama `Anita`, status `member`, total belanja `250000`, jumlah barang `4`, dan kode promo `HEMAT10`.

1. **Tahap masukan.** Program menampilkan judul sistem, lalu meminta lima data melalui `input()`: nama pelanggan, status pelanggan, total belanja, jumlah barang, dan kode promo. Data total belanja dan jumlah barang dikonversi ke `int` karena akan dihitung, sedangkan status dan kode promo dinormalisasi dengan `strip().lower()` dan `strip().upper()` agar perbedaan huruf besar-kecil tidak menyebabkan kesalahan validasi.

2. **Tahap penetapan acuan.** Program menyiapkan konstanta toko, yaitu batas minimum belanja Rp200.000, minimum 3 barang, besar diskon 10 persen, serta daftar kode promo yang tersedia.

3. **Tahap validasi.** Operator perbandingan `>=` dan `==` digunakan untuk memeriksa syarat belanja, syarat jumlah barang, dan status member. Operator keanggotaan `in` serta `not in` memeriksa apakah kode promo pelanggan terdaftar. Seluruh pemeriksaan ini menghasilkan nilai boolean.

4. **Tahap penggabungan kondisi.** Operator logika `and` menggabungkan status member dengan syarat belanja untuk menentukan kelayakan diskon, dan menggabungkan ketersediaan promo dengan syarat jumlah barang untuk menentukan kelayakan promo. Operator `or` menentukan status pelanggan prioritas, sedangkan `not` membalik status member.

5. **Tahap perhitungan.** Apabila pelanggan berhak memperoleh diskon, besar diskon dihitung dengan `total_belanja_3018 * persen_diskon_3018 // 100`, yaitu Rp25.000 pada contoh di atas. Total pembayaran diperbarui menggunakan augmented assignment `-=` menjadi Rp225.000, poin pelanggan ditambah dengan `+=` dan digandakan dengan `*=` bagi member. Rata-rata harga barang diperoleh dari pembagian bulat `//` sebesar Rp56.250, sedangkan sisa pembagian diperoleh dari operator `%`.

6. **Tahap pemeriksaan identitas.** Variabel `salinan_promo_3018` dibuat menunjuk objek list yang sama dengan `daftar_promo_3018`, sedangkan `duplikat_promo_3018` dibuat sebagai objek baru melalui *slicing*. Operator `is`, `is not`, dan `==` kemudian digunakan untuk menunjukkan perbedaan antara identitas objek dan kesamaan nilai.

7. **Tahap penyusunan kode status.** Setiap kondisi yang terpenuhi menyalakan bitnya menggunakan operator OR bitwise `|=`, sehingga terbentuk kode status transaksi `1111` atau 15 dalam desimal. Operator AND `&` dipakai memeriksa bit tertentu, operator XOR `^` membandingkan kode transaksi dengan kode referensi `1011` sehingga menghasilkan `0100` sebagai penanda letak perbedaan, dan operator geser kiri `<<` menggeser seluruh bit satu posisi menjadi `11110` atau 30 dalam desimal.

8. **Tahap keluaran.** Program mencetak data pelanggan, hasil perhitungan, hasil validasi, hak akses pelanggan, serta rincian penggunaan setiap jenis operator secara berurutan sampai penanda `=== SELESAI ===`.

---
## 2.4 Pertanyaan Analisis

**1. Jelaskan operator aritmatika apa saja yang digunakan dan bagaimana operator tersebut membantu menghitung transaksi.**

Operator aritmatika yang digunakan pada program adalah perkalian (`*`), pembagian bulat (`//`), pengurangan (`-`), dan sisa bagi (`%`). Perkalian dan pembagian bulat dipakai bersama untuk menghitung besar diskon melalui ekspresi `total_belanja_3018 * persen_diskon_3018 // 100`, sehingga hasilnya berupa bilangan bulat rupiah. Pengurangan, yang ditulis dalam bentuk `total_bayar_3018 -= diskon_3018`, memotong total belanja dengan besar diskon untuk memperoleh total pembayaran akhir. Pembagian bulat juga dipakai untuk menghitung rata-rata harga per barang, sedangkan operator `%` menghitung sisa pembagian total pembayaran terhadap jumlah barang, yaitu nilai yang tidak dapat terbagi rata di antara barang yang dibeli.

**2. Jelaskan perbedaan operator `and`, `or`, dan `not` berdasarkan penerapannya pada program.**

Operator `and` menghasilkan `True` hanya jika seluruh kondisi bernilai benar. Pada program, `dapat_diskon_3018 = is_member_3018 and syarat_belanja_3018` bernilai `True` hanya apabila pelanggan berstatus member sekaligus total belanjanya mencapai batas minimum; gagal salah satu saja, diskon tidak diberikan. Operator `or` menghasilkan `True` apabila minimal satu kondisi benar, seperti pada `pelanggan_prioritas_3018 = dapat_diskon_3018 or dapat_promo_3018`, sehingga pelanggan tetap dianggap prioritas walaupun hanya memenuhi salah satu keuntungan. Operator `not` bersifat unary dan membalik nilai kebenaran suatu kondisi, seperti pada `bukan_member_3018 = not is_member_3018` yang bernilai `True` justru ketika pelanggan bukan member.

**3. Jelaskan fungsi operator `in` dan `not in` pada pemeriksaan kode promo.**

Operator `in` memeriksa apakah sebuah nilai terdapat di dalam objek sekuens. Pada program, `kode_promo_3018 in daftar_promo_3018` menelusuri list `["HEMAT10", "HEMAT20", "GRATISONGKIR"]` dan menghasilkan `True` bila kode yang dimasukkan pelanggan terdaftar. Operator `not in` merupakan kebalikannya, yaitu menghasilkan `True` bila kode promo tidak ditemukan pada daftar. Penggunaan kedua operator ini membuat pemeriksaan jauh lebih ringkas dibandingkan menuliskan perbandingan `==` satu per satu untuk setiap kode promo, dan daftar promo tetap mudah diperbarui tanpa mengubah logika program.

**4. Apa perbedaan antara `is` dan `==` pada Python?**

Operator `==` membandingkan **nilai atau isi** dari dua objek, sedangkan `is` membandingkan **identitas objek**, yaitu apakah kedua variabel benar-benar menunjuk ke objek yang sama di dalam memori. Pada program, `duplikat_promo_3018` dibuat dengan menyalin isi `daftar_promo_3018` melalui *slicing*, sehingga `duplikat_promo_3018 == daftar_promo_3018` bernilai `True` karena isinya persis sama, tetapi `duplikat_promo_3018 is daftar_promo_3018` bernilai `False` karena keduanya adalah dua objek list yang berbeda. Sebaliknya `salinan_promo_3018 = daftar_promo_3018` hanya membuat nama baru untuk objek yang sama, sehingga `salinan_promo_3018 is daftar_promo_3018` bernilai `True`.

**5. Jelaskan bagaimana operator bitwise digunakan untuk menentukan hak akses pelanggan.**

Setiap kondisi pelanggan diwakili oleh satu bit: `0001` untuk status member, `0010` untuk belanja minimal Rp200.000, `0100` untuk jumlah barang minimal 3, dan `1000` untuk ketersediaan kode promo. Kode status transaksi dibangun dengan operator OR (`|=`): setiap kondisi yang terpenuhi menyalakan bit yang bersesuaian, sehingga seluruh status pelanggan tersimpan dalam satu bilangan bulat saja. Untuk memeriksa kondisi tertentu digunakan operator AND (`&`) dengan bit penanda, misalnya `kode_status_3018 & bit_member_3018`; hasilnya bukan nol apabila bit tersebut menyala. Operator XOR (`^`) dipakai membandingkan kode transaksi dengan kode referensi, di mana bit yang bernilai 1 pada hasil menunjukkan letak perbedaan status antara keduanya. Hak akses gratis ongkir ditentukan dengan menggabungkan dua bit menjadi *mask* `1100` lalu memeriksanya dengan AND, sehingga akses hanya diberikan bila kedua kondisi terpenuhi sekaligus.

**6. Jelaskan alur program mulai dari input hingga menghasilkan total pembayaran dan status promo.**

Program dimulai dengan menampilkan judul sistem, kemudian menerima lima masukan pengguna berupa nama, status, total belanja, jumlah barang, dan kode promo. Masukan numerik dikonversi ke integer agar dapat dihitung. Selanjutnya program membandingkan total belanja dengan batas minimum dan jumlah barang dengan minimum barang menggunakan operator perbandingan, serta memeriksa kode promo dengan operator keanggotaan. Hasil-hasil boolean tersebut digabungkan dengan operator logika untuk menentukan apakah pelanggan berhak memperoleh diskon dan promo. Apabila berhak, besar diskon dihitung dengan operator aritmatika, lalu total pembayaran diperbarui menggunakan augmented assignment `-=`, dan rata-rata harga barang dihitung dengan pembagian bulat. Seluruh kondisi kemudian diterjemahkan menjadi kode status biner melalui operator bitwise untuk menentukan hak akses pelanggan. Terakhir, program menampilkan data pelanggan, hasil perhitungan, hasil validasi, hak akses, serta rincian hasil setiap operator.

**7. Identifikasi minimal satu kondisi dalam program yang menghasilkan `True` dan satu kondisi yang menghasilkan `False`.**

Dengan masukan nama `Raldo Restanggi Alfarizy`, status `member`, total belanja `250000`, jumlah barang `4`, dan kode promo `HEMAT10` seperti pada output di atas:
- Kondisi `syarat_belanja_3018 = total_belanja_3018 >= 200000` menghasilkan **`True`**, karena 250000 lebih besar dari batas minimum belanja. Akibatnya bit `0010` menyala pada kode status dan pelanggan berhak memperoleh diskon.
- Kondisi `bukan_member_3018 = not is_member_3018` menghasilkan **`False`**, karena pelanggan berstatus member sehingga `is_member_3018` bernilai `True` dan dibalik oleh operator `not`.

Sebagai pembanding, apabila masukan diganti dengan status `nonmember`, total belanja `150000`, jumlah barang `2`, dan kode promo `DISKON99`, maka `promo_tersedia_3018` menghasilkan `False` karena kode tersebut tidak terdapat pada daftar promo toko, dan kode status transaksi bernilai `0000` sehingga seluruh hak akses pelanggan tertutup.

---
# BAB III
# KESIMPULAN

Berdasarkan praktikum yang telah dilakukan, dapat disimpulkan bahwa operator merupakan komponen dasar yang menentukan bagaimana sebuah program mengolah data dan mengambil keputusan. Setiap jenis operator memiliki peran yang berbeda namun saling melengkapi: operator aritmatika menangani perhitungan nilai transaksi, operator perbandingan mengubah fakta numerik menjadi nilai boolean, operator logika menggabungkan beberapa syarat menjadi satu keputusan, operator penugasan memperbarui nilai variabel secara ringkas, operator keanggotaan mempermudah pencarian data pada sekuens, operator identitas membedakan objek yang sama dari objek yang sekadar bernilai sama, dan operator bitwise memungkinkan banyak kondisi disimpan serta diperiksa dalam satu bilangan.

Dalam konteks studi kasus sistem transaksi toko, penggabungan seluruh operator tersebut membentuk satu alur program yang utuh, mulai dari penerimaan data pelanggan, validasi syarat diskon dan promo, perhitungan total pembayaran, hingga penentuan hak akses melalui kode status biner. Praktikum ini juga menegaskan pentingnya pemilihan operator yang tepat, misalnya penggunaan `//` untuk menjaga hasil perhitungan rupiah tetap bulat, serta penggunaan `is` yang tidak dapat dipertukarkan dengan `==` karena keduanya membandingkan hal yang berbeda.

---
## DAFTAR PUSTAKA

[1] Python Software Foundation, "The Python Language Reference: Expressions," Python Documentation, 2026. [Online]. Tersedia: https://docs.python.org/3/reference/expressions.html [Diakses 17 September 2026].

[2] Python Software Foundation, "Built-in Types," Python Documentation, 2026. [Online]. Tersedia: https://docs.python.org/3/library/stdtypes.html [Diakses 17 September 2026].

[3] Tim Asisten Praktikum, "Modul Praktikum 3: Operator Python," Departemen Informatika, Fakultas Teknologi Informasi, Universitas Andalas, 2026.